utilities/virt.py

```diff
@@ -2019,19 +2020,22 @@ def vm_instance_from_template(
 
 
 @contextmanager
-def node_mgmt_console(node, node_mgmt):
+def node_mgmt_console(
+    node: Node,
+    node_mgmt: Literal["cordon", "drain"],
+    timeout: int | None = None,
+) -> Generator[None, None, None]:
     try:
         LOGGER.info(f"{node_mgmt.capitalize()} the node {node.name}")
-        extra_opts = "--delete-emptydir-data --ignore-daemonsets=true --force" if node_mgmt == "drain" else ""
-        run(
-            f"nohup oc adm {node_mgmt} {node.name} {extra_opts} &",
-            shell=True,
-        )
+        cmd = ["oc", "adm", node_mgmt, node.name]
+        if node_mgmt == "drain":
+            cmd += ["--delete-emptydir-data", "--ignore-daemonsets=true", "--force"]
+        Popen(args=cmd, stdout=DEVNULL, stderr=DEVNULL)
         yield
     finally:
         LOGGER.info(f"Uncordon node {node.name}")
-        run(f"oc adm uncordon {node.name}", shell=True)
-        wait_for_node_schedulable_status(node=node, status=True)
+        run(["oc", "adm", "uncordon", node.name], check=True, timeout=timeout)
+        wait_for_node_schedulable_status(node=node, status=True, timeout=timeout or TIMEOUT_5MIN)
```

Comment:
With your changes this function becomes very messy, please keep it untouched
We have many places where we use this function and it always worked fine with the default timeout. 

Please double check if you still have this problem, if for some reason the default is not enough for you - just add it to the line 
`wait_for_node_schedulable_status(node=node, status=True, timeout=TIMEOUT_5MIN)`